shard.go

```diff
@@ -129,8 +130,10 @@ func (s *cacheShard) set(key string, hashedKey uint64, entry []byte) error {
 		}
 	}
 
-	if oldestEntry, err := s.entries.Peek(); err == nil {
-		s.onEvict(oldestEntry, currentTimestamp, s.removeOldestEntry)
+	if !s.cleanEnabled {
```

Comment:
Could you add a test for this?

Reply:
Thank you for your reply, I added testing and then added `if !s.cleanEnabled` when Append